aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -285,7 +287,7 @@ public static ModifyDbInstanceRequest modifyDbInstanceRequest(
                 .caCertificateIdentifier(desiredModel.getCACertificateIdentifier())
                 .copyTagsToSnapshot(desiredModel.getCopyTagsToSnapshot())
                 .dbInstanceClass(desiredModel.getDBInstanceClass())
-                .dbInstanceIdentifier(desiredModel.getDBInstanceIdentifier())
+                .dbInstanceIdentifier(currentIdentifier)
```

Comment:
I wonder how customers can rename DbInstance. I do not see `NewDBInstanceIdentifier` attribute in schema. Then the only way is by replacement after updating `DBInstanceIdentifier` then CFN will detect it is create only then delete and create new instance with same attribute. Is this behavior on intention or I am missing something.

Reply:
That's a good point. I'll get it fixed. TY!

Reply:
Just heads up there is a loop can happen. DescribeDBInstance will return new `DBInstanceIdentifier` which will be detected as drift from source stack, Then CloudFormation will replace again to achieve old `DBInstanceIdentifier` but modify will apply `NewDBInstanceIdentifier` again and it generate loop.

Reply:
Noted and fixed.